windows.js

```diff
@@ -0,0 +1,62 @@
+const execa = require('execa');
+
+const isOn = () => 
+	execa.shell('ping google.com -n 1 -w 1000').then(() => true).catch(() => false);
+
+const device = () => 
+	execa.shell('netsh interface show interface').then((stdout) => {
+		let data = new Array(stdout);
+		data = data[0].stdout.split(' ');
+		
+		return data[data.length - 1].trim();
+	});
+
+
+const toggleDevice = (turnOn = null) => {
+	execa.shell('netsh interface show interface').then((stdout) => {
+		let data = new Array(stdout);
+		data = data[0].stdout.split(' ');
+		
+		let deviceInterface = data[data.length - 1].trim().toLowerCase();
+
+		if(turnOn != null) 
+			return execa.shell(`netsh interface set interface "${deviceInterface}" ${(turnOn) ? 'enabled' : 'disabled'}`)
+					.catch(err => {throw new Error('Error: Run this with administrator privilegies!')});
+		
+		return isOn().then(isOn => {
+			execa.shell(`netsh interface set interface "${deviceInterface}" ${(isOn) ? 'disabled' : 'enabled'}`)
+			.catch(err => {throw new Error('Error: Run this with administrator privilegies!')});
+		});
+	});
+};
+
+const toggle = (turnOn) => {
+	if (typeof turnOn === 'boolean') {
+		return new Promise((resolve) => resolve(toggleDevice(turnOn)));
+	}
+
+	return new Promise((resolve) => resolve(toggleDevice()));
```

Comment:
@hendrysadrak That is also incorrect. No need to double-resolve the promise. You could just return `toggleDevice()` directly. Actually, it seems this function whole function is totally moot. You could just use `toggleDevice()` directly.